proposals/12020-model-registry-integration/README.md

```diff
@@ -0,0 +1,432 @@
+providing data scientists with a simple interface for model registration through the KFP SDK. This enhancement bridges
+the gap between KFP's artifact storage and Model Registry's cataloging capabilities, creating a unified model management
+experience.
+
+## Motivation
+
+### Current State and Limitations
+
+Kubeflow Pipelines currently maintains its own artifact store for pipeline outputs, which provides basic storage
+functionality but lacks advanced cataloging and versioning features. Models can only be added to this store through:
+
+1. **Pipeline execution outputs** (e.g., `dsl.Output[dsl.Model]`)
+2. **Importer components** for external model registration
+
+This approach has several limitations:
+
+- **No centralized cataloging**: Models are scattered across pipeline runs without unified discovery
+- **Limited versioning**: No structured version management or lineage tracking
+- **Tool-specific isolation**: Models created in KFP are not discoverable by other Kubeflow components
+- **Manual registration overhead**: Users must manually register models outside of pipeline workflows
+
+### Integration Benefits
+
+By enabling seamless Model Registry integration, users will experience:
+
+- **Centralized model catalog**: Single source of truth for all models across the Kubeflow ecosystem
+- **Versioning**: Structured version management with lineage tracking
+- **Cross-tool discovery**: Models become discoverable by other Kubeflow components (e.g. KServe)
+- **Simplified workflows**: One-line model registration within pipeline components
+- **Enhanced governance**: Better model lifecycle management and compliance tracking
+
+### Current Workarounds
+
+Without this integration, users must implement complex workarounds:
+
+1. Add Model Registry infrastructure details in the pipeline (e.g. URLs) through hardcoded values, pipeline input
+   parameters, or mounting a Kubernetes `ConfigMap`/`Secret`.
+1. Mount a Kubernetes `Secret` to access the token.
+1. Know the KFP standards for registering models in Model Registry (e.g. `model_source_kind="kfp"`).
+
+### Goals
+
+1. **Seamless SDK Integration**: Provide a simple, API for model registration within KFP components
+2. **Infrastructure Abstraction**: Hide Model Registry connection details and authentication from user code
+3. **Standardized Metadata**: Automatically populate KFP-specific metadata for proper model lineage tracking
+4. **Error Resilience**: Provide configurable error handling to prevent pipeline failures due to registration issues
+5. **Multi-tenancy Support**: Enable namespace-specific configuration for isolated deployments
+
+### Non-Goals
+
+1. Implement KFP-specific RBAC controls over Model Registry APIs, such as model allowlists for version registration.
+
+## Proposal
+
+### SDK User Experience
+
+The proposed integration introduces a `register()` method on KFP Model artifacts, providing a clean interface for model
```

Comment:
understanding here, this `.register()` is available to the DSP author, so can also choose to selectively register it or not depending on the logic of the pipeline itself, sgtm

Reply:
That's right! The user could choose to do the registration in a separate evaluation component from the one that generated the model.

Reply:
> That's right! The user could choose to do the registration in a separate evaluation component from the one that generated the model.

thanks for the confirmation!